src/chunk_index.c

```diff
@@ -507,26 +507,29 @@ chunk_index_create_all(int32 hypertable_id, Oid hypertable_relid, int32 chunk_id
 
 static int
 chunk_index_scan(int indexid, ScanKeyData scankey[], int nkeys,
-				 tuple_found_func tuple_found, void *data, LOCKMODE lockmode)
+				 tuple_found_func tuple_found, tuple_filter_func tuple_filter, void *data, LOCKMODE lockmode)
 {
 	Catalog    *catalog = catalog_get();
 	ScannerCtx	scanCtx = {
 		.table = catalog->tables[CHUNK_INDEX].id,
-		.index = catalog->tables[CHUNK_INDEX].index_ids[indexid],
-		.scantype = ScannerTypeIndex,
 		.nkeys = nkeys,
 		.scankey = scankey,
 		.tuple_found = tuple_found,
+		.filter = tuple_filter,
 		.data = data,
 		.lockmode = lockmode,
 		.scandirection = ForwardScanDirection,
 	};
 
+
+	if (indexid != INVALID_INDEXID)
```

Comment:
Can we create a macro or a function that looks something like:

```
CATALOG_INDEX(catalog, tableid, indexid) \
    (indexid == INVALID_INDEX ? InvalidOid : (catalog)->tables[tableid].index_ids[indexid])
```
Then we can set the index when defining the scan context:

```
.index = CATALOG_INDEX(catalog, tableid, indexid),
```

and avoid the `if` clause following scan context initialization. This seems like a common-enough thing to warrant that.